packages/http/src/validator/validators/security/index.ts

```diff
@@ -40,15 +41,15 @@ export const validateSecurity: ValidatorFn<Pick<IHttpOperation, 'security'>, Pic
   element,
   resource,
 }) => {
-  const securitySchemes = resource.security;
-
-  if (securitySchemes && isNonEmpty(securitySchemes)) {
-    return pipe(
-      getValidationResults(securitySchemes, element),
-      Either.mapLeft<NonEmptyArray<IPrismDiagnostic>, NonEmptyArray<IPrismDiagnostic>>(e => [getWWWAuthHeader(e)]),
-      Either.map(() => element)
-    );
-  } else {
-    return Either.right(element);
-  }
+  return pipe(
+    Option.fromNullable(resource.security),
+    Option.chain(securitySchemes => pipe(securitySchemes, Option.fromPredicate(isNonEmpty))),
+    Option.map(securitySchemes => pipe(
+        getValidationResults(securitySchemes, element),
+        Either.mapLeft<NonEmptyArray<IPrismDiagnostic>, NonEmptyArray<IPrismDiagnostic>>(e => [getWWWAuthHeader(e)]),
+        Either.map(() => element)
+      )
+    ),
+    Option.fold(() => Either.right(element), identity as <T>(a: T) => T),
```

Comment:
`lodash`'s `identity` typings kind of suck. We can replace with:
```suggestion
    Option.fold(() => Either.right(element), r => r),
```